hacks/generateOptimizeSidebars.js

```diff
@@ -74,22 +89,70 @@ function buildLi(li) {
   }
 }
 
-// The root of the sidebar menu.
-const root = document.querySelector(".theme-doc-sidebar-menu");
-
 // Generates the sidebar definitions for the entire menu.
 function buildSidebars() {
+  // The root of the sidebar menu.
+  const root = document.querySelector(".theme-doc-sidebar-menu");
+  if (root === null) {
+    throw new Error("Unable to find sidebar menu root.");
+  }
+
   const items = Array.from(root.querySelectorAll(":scope > li"));
 
   const mappedItems = items.map(buildLi);
 
-  return `{
-    "Components": [
+  return `[
       ${mappedItems.join("\n")}
-    ]
+    ]`;
+}
+
+// Identifies configuration strategy based on current URL.
+function identifyStrategyFromPath() {
+  const path = window.location.pathname;
+  const strategyKey = Object.keys(strategies).find((strategyPath) =>
+    path.startsWith(strategyPath)
+  );
+
+  if (strategyKey === undefined) {
+    throw new Error(
+      "I'm not sure how to generate sidebars for this URL. Are you sure you're on the correct page?"
+    );
   }
-  `;
+  strategy = strategies[strategyKey];
 }
 
-// Copies the generated sidebar definitions to the clipboard.
-copy(buildSidebars());
+// Expands all sidebar items (so that we can walk them all in the DOM).
+async function expandAllSidebarItems() {
+  const count = () =>
+    document.querySelectorAll(".menu__link--sublist[aria-expanded=false]")
+      .length;
+  const expand = () =>
+    document
+      .querySelectorAll(".menu__link--sublist[aria-expanded=false]")
+      /** @ts-ignore */
+      .forEach((x) => x.click());
+  async function expandAll() {
+    while (count() > 0) {
+      expand();
+      await new Promise((resolve) => setTimeout(resolve, 500));
+    }
+  }
+  await expandAll();
+}
+
+// Store a reference to the copy function globally, because sometimes it's not available from within a promise.
+/** @ts-ignore */
+const copyToClipboard = copy;
+
```

Comment:
ts-ignored because `copy` is not fully supported across all browsers. It is supported by anything running chromium though, and I'm just going to assume people are going to use one of those.

(I did experiment with using the fully-supported Clipboard API instead of `copy`...however it doesn't work from within the browser dev tools 😓 which is a dealbreaker.)

Reply:
Is this worth documenting somewhere or including inline?

Reply:
Yes, I'll add to the comment!

Reply:
updated in most recent commit